src/nrncvode/netcvode.cpp

```diff
@@ -5522,8 +5507,8 @@ static int trajec_buffered(NrnThread& nt, int bsize, IvocVect* v, double* pd,
 {
   int err = 0; //success
   if (bsize > 0) {
-    if (v->buffer_size() < bsize) {
-      v->buffer_size(bsize);
+    if (v->size() < bsize) {
+      v->resize(bsize);
```

Comment:
Now I'm confused. the gnu implementation distinguished between buffer_size and capacity (the former being an allocated space that would persist until capacity became so large the buffer had to be reallocated ( People would sometimes allocate a buffer_size large enough so as not to have to worry about pointers to the vector data becoming invalid.  I didn't look closely at this particular code, but I presume the size() might be 0

Reply:
`buffer_size` is compatible with `std::vector` through `std::vector->capacity()` and `buffer_size(n)` through `std::vector->reserve(n)`. That's why I said previously `doubleVec->capacity()` should be looked at as a misnomer. We now have:

```cpp
int IvocVect::buffer_size() {
	return vec_.capacity();
}

void IvocVect::buffer_size(int n) {
     vec_.reserve(n);
}
```

Coming back to this change, it took quite some time to figure it out. The reason it should be now `size` is because we are ensuring that the vector can be written to on a trajectory return. For `std::vector` we can allocate a big enough space, however we cannot use it as a buffer as compared to `doubleVec`: the memory would be accessible, but the length cannot directly be manipulated. Here we are on a trajectory return and we have a specific size to write to, that's why we consequently use `std::vector->resize()`. The `resize()` would still perform a re-allocation behind the hood if needed.

Vector aside, do we have a valid case today where `bsize` is not greater than 0? Looking at the code coverage, this isn't tested in our CI:
![Screenshot 2021-02-15 at 09 31 55](https://user-images.githubusercontent.com/46521150/107922673-a7f7fe80-6f70-11eb-9a44-445d62f259c9.png)

Reply:
Thanks. And I don't think there are any examples  of a buffer size request of 0.

Reply:
Suspected as much. Thanks